### DISCUSSION TITLE
I need help modifing the path from a subapplication.

### DISCUSSION THREAD

**Discussion by gumbarros** [dotnet/yarp#1844] [Q&A] [answered]
# My Scenario

I have the following IIS site:

www.site.com

Calling www.site.com/Login redirects correctly to the page in my subapplication, the browser bar shows www.site.com/Login.

But, calling www.site.com, redirects to www.site.com/subapplication/Login, it also works, but I want www.site.com to show www.site.com/Login in the browser bar.

# IIS Structure

www.site.com is a .NET Core site with YARP

www.site.com/subapplication/Login is a .NET Framework site

# appsettings.json
```
  "ReverseProxy": {
    "Routes": {
      "fallbackRoute": {
        "ClusterId": "fallbackCluster",
        "Order": "1",
        "Match": {
          "Path": "{**catch-all}"
        },
        "Transforms": [
          { "PathRemovePrefix": "/web" }
          { "RequestHeaderOriginalHost": "true" }
        ]
      }
    },
    "Clusters": {
      "fallbackCluster": {
        "Destinations": {
          "fallbackApp": {
            "Address": "https://dansaleswebdev.danone.com.br/dev/web/"
          }
        }
      }
    }
  }
```

# What I want

www.site.com redirect to www.site.com/Login instead of www.site.com/subapplication/Login 

I already tried to add the [PathRemovePrefix transformation](https://microsoft.github.io/reverse-proxy/articles/transforms.html#pathremoveprefix) to remove /subapplication but it not worked.

**Comment by Tratcher** [comment, marked as answer]
See https://github.com/microsoft/reverse-proxy/issues/1548.

You either need to change the framework app to generate the public urls, or you need to re-write the location header on the response.

**Reply by gumbarros** [reply]
Thanks for the response, to re-write the location header on the response is used YARP transformations or a web.config URL rewrite?

I also using SystemWebAdapters on .NET Framework, I don't know if you are related to the project, but is AddProxySupport method with UseForwardedHeaders=true supposed to fix the issue?

**Reply by Tratcher** [reply]
> Thanks for the response, to re-write the location header on the response is used YARP transformations or a web.config URL rewrite?

You can use either.

> I also using SystemWebAdapters on .NET Framework, I don't know if you are related to the project, but is AddProxySupport method with UseForwardedHeaders=true supposed to fix the issue?

Yes, we also help with that project. We ran into a similar problem there. IIS doesn't allow changing the root path when it would conflict with another site.

**Reply by gumbarros** [reply]
I tried the entire morning using a URL rewrite via IIS and web.config but no success 🙁. How can I achieve this via YARP transformations?

**Reply by Tratcher** [reply]
https://github.com/microsoft/reverse-proxy/issues/1548 shows part of the code you need.

**Reply by gumbarros** [reply]
Solved in SystemWebAdapters [#174](https://github.com/dotnet/systemweb-adapters/issues/174)
